src/zmq.cpp

```diff
@@ -161,7 +161,12 @@ void *zmq_ctx_new (void)
 
     //  Create 0MQ context.
     zmq::ctx_t *ctx = new (std::nothrow) zmq::ctx_t;
-    alloc_assert (ctx);
+    if (ctx) {
+        if (!ctx->valid ()) {
+            delete ctx;
+            return NULL;
```

Comment:
I'm not sure about this - it never returned null, so nobody will be checking for it, which means we might get a flurry of segfaults reports...

Reply:
@bluca The documentation states that:

´´´
The zmq_ctx_new() function shall return an opaque handle to the newly created context if successful. Otherwise it shall return NULL and set errno to one of the values defined below.
´´´

so I would expect good code to already check for null.